server/src/main/kotlin/core/gameflow/Player.kt

```diff
@@ -0,0 +1,35 @@
+package core.gameflow
+
+import core.Card
+
+data class Player(
+    val position: Int,
+    val stack: Int,
+    val holeCards: List<Card>,
+    val chipsInPot: Int,
+    val currentBet: Int,
+    val folded: Boolean = false
+) {
+    fun withCards(newCards: List<Card>): Player {
+        return this.copy(holeCards = newCards)
+    }
+
+    fun afterFold(): Player {
+        return this.copy(folded = true)
+    }
+
+    fun afterAllIn(): Player {
+        return this.copy(currentBet = this.currentBet + this.stack, stack = 0)
+    }
+
+    fun withBet(betSize: Int): Player {
+        if (betSize > this.stack)
+            throw NotEnoughChipsException("Bet size of $betSize larger than stack size ${this.stack}")
+
+        return this.copy(currentBet = this.currentBet + betSize, stack = this.stack - betSize)
+    }
+
+    fun isAllIn(): Boolean {
+        return stack == 0 && currentBet != 0 && !folded
+    }
+}
```

Comment:
Maybe we should add 'afterCall(betSize: Int)' method that is similar to withBet except for making player all-in in case the bet size is higher than the player's stack?

Reply:
I don't know if this will be necessary, we shall see later I guess

Reply:
Imagine a scenario where you want to apply 'call' action to given player. With only 'withBet' method we would first need to check if the bet size doesn't exceed the stack (or even worse, use try-catch expression) and apply proper method (withBet / afterAllIn). Suggested additional method would delegate this logic to Player's class (where I think it should belong).

Reply:
Okay I agree